app/src/main/java/com/h/pixeldroid/FeedRecyclerViewAdapter.kt

```diff
@@ -0,0 +1,84 @@
+package com.h.pixeldroid
+
+import android.content.Context
+import android.graphics.Typeface
+import android.graphics.drawable.Drawable
+import android.util.DisplayMetrics
+import androidx.recyclerview.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.view.WindowManager
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.annotation.Dimension
+import com.h.pixeldroid.models.Post
+import com.h.pixeldroid.utils.ImageConverter.Companion.setDefaultImage
+import com.h.pixeldroid.utils.ImageConverter.Companion.setImageViewFromURL
+import com.h.pixeldroid.utils.ImageConverter.Companion.setRoundImageFromURL
+
+/**
+ * [RecyclerView.Adapter] that can display a list of [Post]s
+ */
+class FeedRecyclerViewAdapter(
+    private val posts: List<Post>,
```

Comment:
posts shouldn't be added in the constructor, you should create the adapter when the fragment is created (not in the callback from the network), then add the posts in the callback. (See https://github.com/H-PixelDroid/PixelDroid/blob/1fa4fc3e0d6eab94b036e50f81b6989fc8c55cc7/app/src/main/java/com/h/pixeldroid/fragments/NotificationsFragment.kt#L55 in the notifications branch) 

This will allow having a loading spinner, having the pull to refresh animation work even while loading, and not having an inflation error in the logs